### ISSUE TITLE
Integrate REST API documentation in Docusaurus website

### ISSUE THREAD

**Issue by wborn** [openremote/Documentation#6]
The [docusaurus-plugin-openapi-docs](https://docusaurus-openapi.tryingpan.dev/) can be used for generating REST API documentation using the OpenAPI spec.

For a PoC using this plugin, see the changes in my [fork](https://github.com/wborn/docusaurus.maindrain.net) and its respective [website](https://docusaurus.maindrain.net/).
E.g. click on "REST API" in the top bar or expand the "rest-api" section in the index on the left.

While making the PoC I ran into a few issues that still need to be resolved before this can be integrated:

- [x] The current OpenAPI spec is invalid which need to be fixed before it can be used for generating documentation.
For the PoC I manually corrected this but it would be better if the code is updated so a valid spec is generated.
You can also see what is invalid in [Swagger UI](https://demo.openremote.io/swagger/#/) if you scroll down to the bottom and click the [INVALID button](https://validator.swagger.io/validator/debug?url=https%3A%2F%2Fdemo.openremote.io%2Fapi%2Fmaster%2Fopenapi.json).
- [x] The API operations are generated using the method names which doesn't look good. I.e. instead of `doProtocolAssetDiscovery` it would be nicer if the operation is listed as "Do protocol asset discovery". This can be fixed by annotating all REST methods with a proper [`@Operation` annotation](https://www.baeldung.com/swagger-operation-vs-apiresponse#operation).
- [x] It would be nice if the operations in the "rest-api" section on the left are grouped. If this is not easy to do we should remove it and just link to the URL as done in the top bar.
- [x] The language examples ([top right](https://docusaurus.maindrain.net/docs/rest-api/do-protocol-asset-discovery)) should be reordered a bit so more commonly used examples are shown first (i.e. Java, Python, NODEJS, curl ...)
- [x] Using API example requests on demo.openremote.io is currently not possible because of CORS errors. For this to be fixed demo.openremote.io should allow requests from docs.openremote.io by setting the `Access-Control-Allow-Origin` response header. See also the [Swagger UI CORS documentation](https://swagger.io/docs/open-source-tools/swagger-ui/usage/cors/).
- [x] The readme needs to be updated on how to generates REST API documentation.

**Comment by pankalog** [comment]
@wborn the demo instance does not have the fix introduced in https://github.com/openremote/openremote/pull/1320 which fixes all (but two) of the OpenAPI scheme issues, so I'm guessing that the first bullet point can be resolved. The two errors are minor, and you can test it out on your local OR instance.

**Comment by wborn** [comment]
> does not have the fix introduced in https://github.com/openremote/openremote/pull/1320

Yes we also just noticed most spec issues are fixed on the staging demo. :+1:

**Comment by wborn** [comment]
I did some more testing and https://github.com/openremote/openremote/pull/1320 fixes most  spec issues. The only issue remaining was that the Docusaurus plugin cannot handle `OAuthFlows` without scopes. So after adding a scope to the  `authorizationCode` flow it can now generate the REST API documentation without manual changes using the OpenRemote OpenAPI spec. 

With the updated spec it now also shows the allowed enum values. As a test I also added  `@Operation` annotations to all Asset operations. See the [updated PoC](https://docusaurus.maindrain.net/docs/rest-api/query-assets/).
